json_test.go

```diff
@@ -0,0 +1,113 @@
+package ff
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestJSONParser(t *testing.T) {
+	type want struct {
+		s   string
+		i   int
+		b   bool
+		d   time.Duration
+		err string
+	}
+
+	for _, testcase := range []struct {
+		name string
+		args []string
+		file string
+		want want
+	}{
+		{
+			name: "empty input",
+			args: []string{},
+			file: `{}`,
+			want: want{d: time.Second},
+		},
+		{
+			name: "basic KV pairs",
+			args: []string{},
+			file: `{"s": "s", "i": 10, "b": true, "d": "5s"}`,
+			want: want{"s", 10, true, 5 * time.Second, ""},
+		},
+		{
+			name: "Key with array of values",
+			args: []string{},
+			file: `
+                {
+                    "s": ["t", "s"],
+                    "i": ["11", "10"],
+                    "b": [false, true],
+                    "d": ["10m", "5s"]
+                }
+            `,
+			want: want{"s", 10, true, 5 * time.Second, ""},
+		},
+		{
+			name: "bad JSON file",
+			args: []string{},
+			file: `{`,
+			want: want{d: time.Second, err: "error parsing JSON config"},
```

Comment:
Is there a way to improve this, to leverage the ParseError type, so you're not comparing strings?

Reply:
Which ParseError? You mean the errors in encoding/json? For a test, I'm inclined to be happy with just a quick and dirty string test.